.github/workflows/docker-build-push.yml

```diff
@@ -68,19 +68,26 @@ jobs:
       # download data
       - name: Download data
         run: bash scripts/download-data.sh
-      
+
       - name: Render all pages to html
         run: |
           docker run \
           --mount type=bind,target=/home/rstudio,source=$PWD \
           ccdl/refinebio-examples \
           snakemake --cores 2 --forceall
 
-      - name: Commit changed html
+      # If we are on the staging branch, do not publish to github pages
+      - name: Commit changed html back to staging branch
+        if: github.ref == 'refs/heads/staging'
         run: |
           git add -A
-          git commit -m 'Render html' || echo "No changes to commit"
-          git push origin gh-pages || echo "No changes to push"
-
-
+          git commit -m 'Render html, do not publish' || echo "No changes to commit"
+          git push origin gh-pages-stages || echo "No changes to push"
 
+      # If we are on the master branch, publish to github pages!
+      - name: Commit changed html to public facing gh-pages
+        if: github.ref == 'refs/heads/master'
+        run: |
+          git add -A
+          git commit -m 'Render html and publish' || echo "No changes to commit"
+          git push origin gh-pages || echo "No changes to push"
```

Comment:
This is only half the battle... We also need to change line 28 to checkout the correct branch, depending on `github.ref`. This probably means passing `github.ref` in as an ENV variable in the `Checkout pages branch and sync with changes` step, then using that in an if statement for which branch to checkout.

You could use the same logic to collapse the last two steps here, but I don't think that is necessary.